Source/Scene/Material.js

```diff
@@ -895,12 +913,18 @@ function createTexture2DUpdateFunction(uniformId) {
           promise = resource.fetchImage();
         }
 
-        when(promise, function (image) {
-          material._loadedImages.push({
-            id: uniformId,
-            image: image,
-          });
-        });
+        when(
+          promise,
+          function (image) {
+            material._loadedImages.push({
+              id: uniformId,
+              image: image,
+            });
+          },
+          function () {
+            material._textures[uniformId] = material._defaultTexture;
```

Comment:
Double checking here - how does this interact with the code above? Does it recognize that the texture changed and destroy the previous texture? Is there a way to fit this cleanly into that system?

Reply:
Ah I see that the code in `update` was handling the destroy for this. Can it somehow be handled by `createTexture2DUpdateFunction` instead?